### PR TITLE
Fix Shadow demo misalignment with MQTT demo helper.

### PR DESCRIPTION
**Pull request by lundinc2**
Fix Shadow demo misalignment with MQTT demo helper.

Description
-----------
Moving the shadow helper to a common MQTT helper flipped the success and fail values.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by yourslab — DISMISSED** [review]

File: demos/device_shadow_for_aws/shadow_demo_main.c [hunk 1]
```diff
@@ -568,7 +568,7 @@ int RunDeviceShadowDemo( bool awsIotMqttMode,
                          void * pNetworkCredentialInfo,
                          const void * pNetworkInterface )
 {
-    int returnStatus = EXIT_SUCCESS;
+    int returnStatus = pdPASS;
```
Comment (aggarw13) [inline demos/device_shadow_for_aws/shadow_demo_main.c:571]: We should use `BaseType_t` as the type for `pdPASS` and `pdFAIL`
```suggestion
    BaseType_t xDemoStatus = pdPASS;
```

File: demos/device_shadow_for_aws/shadow_demo_main.c [hunk 2]
```diff
@@ -760,10 +760,10 @@ int RunDeviceShadowDemo( bool awsIotMqttMode,
...
          * MQTT topic fails or there are failure in parsing the received JSON document,
          * then this demo was not successful. */
-        if( ( lUpdateAcceptedReturn != EXIT_SUCCESS ) || ( lUpdateDeltaReturn != EXIT_SUCCESS ) )
+        if( ( lUpdateAcceptedReturn != pdPASS ) || ( lUpdateDeltaReturn != pdPASS ) )
         {
             LogError( ( "Callback function failed." ) );
-            returnStatus = EXIT_FAILURE;
+            returnStatus = pdFAIL;
         }
     }
```
Comment (aggarw13) [inline demos/device_shadow_for_aws/shadow_demo_main.c:769]: The return value of the demo can be provided as `return ( ( xDemoStatus == pdPASS )? EXIT_SUCCESS : EXIT_FAILURE );`
as the return value should be `EXIT_SUCCESS/EXIT_FAILURE`.

File: demos/device_shadow_for_aws/shadow_demo_main.c [hunk 3]
```diff
@@ -215,12 +215,12 @@ static uint32_t ulClientToken = 0U;
 /**
  * @brief The return status of prvUpdateDeltaHandler callback function.
  */
-static int32_t lUpdateDeltaReturn = EXIT_SUCCESS;
+static int32_t lUpdateDeltaReturn = pdPASS;
```
Comment (aggarw13) [inline demos/device_shadow_for_aws/shadow_demo_main.c:218]:
```suggestion
static BaseType_t xUpdateDeltaReturn = pdPASS;
```

File: demos/device_shadow_for_aws/shadow_demo_main.c [hunk 4]
```diff
@@ -215,12 +215,12 @@ static uint32_t ulClientToken = 0U;
...
 
 /**
  * @brief The return status of prvUpdateAcceptedHandler callback function.
  */
-static int32_t lUpdateAcceptedReturn = EXIT_SUCCESS;
+static int32_t lUpdateAcceptedReturn = pdPASS;
```
Comment (aggarw13) [inline demos/device_shadow_for_aws/shadow_demo_main.c:223]:
```suggestion
static BaseType_t xUpdateAcceptedReturn = pdPASS;
```

**Review by aggarw13 — DISMISSED** [review]

**Review by aggarw13 — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**